Core/Models/PointMassSecondOrder.hpp

```diff
@@ -67,13 +74,23 @@ namespace gtfo{
             const Scalar &dt = parameters.dt;
             const Scalar &mass = parameters.mass;
             const Scalar &damping = parameters.damping;
+            const Scalar &stiffness = parameters.stiffness;
 
             // Update the discrete-time state transition matrices, which are computed using exact discretization
-            const Scalar exponent = std::exp(-damping / mass * dt);
-            Base::A_discrete_ << static_cast<Scalar>(1.0), (static_cast<Scalar>(1.0) - exponent) * mass / damping,
-                static_cast<Scalar>(0.0), exponent;
-            Base::B_discrete_ << (damping * dt - (static_cast<Scalar>(1.0) - exponent) * mass) / (damping * damping),
-                (static_cast<Scalar>(1.0) - exponent) / damping;
+            if (stiffness == 0){
+                const Scalar exponent = std::exp(-damping / mass * dt);
+                Base::A_discrete_ << static_cast<Scalar>(1.0), (static_cast<Scalar>(1.0) - exponent) * mass / damping,
+                    static_cast<Scalar>(0.0), exponent;
+                Base::B_discrete_ << (damping * dt - (static_cast<Scalar>(1.0) - exponent) * mass) / (damping * damping),
+                    (static_cast<Scalar>(1.0) - exponent) / damping;}
+            else {
+            // here we used a first order approximation
+                Base::A_discrete_ << static_cast<Scalar>(1.0), dt,
+                    - (dt * stiffness) / mass, static_cast<Scalar>(1.0) - (damping * dt) / mass;
+                Base::B_discrete_ << static_cast<Scalar>(0.0),
+                    dt/mass;
+                }
+            Base::C_discrete_ << static_cast<Scalar>(0.0), stiffness;
```

Comment:
Relating to my other comment, then `C_discrete_` needs to also depend on the spring's zero position